### PR TITLE
Dry-run mode 

### PR DESCRIPTION
**Pull request by noraabiakar**
Dry-run mode: 
* An implementation of distributed_context that is used to mimic the performance of running an MPI distributed simulation with n ranks.
* Verifiable against an MPI run with the same parameters. 

Implementation: 
* Describe the model on a single domain (tile) and translate it to however many domains we want to mimic using arb::tile and arb::symmetric_recipe. This allows us to know the exact behavior of the entire system by only running the simulation on a single node.
* Mimic communication between domains using arb::dry_run_context

Example: 
* dryrun in example/ is a verifiable example of using dry-run mode with mc_cells

Other:
* Documentation of dry-run mode 
* unit test for dry_run_context

### CODE REVIEW

File: example/dryrun/dryrun.cpp [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                 << s.ncomp << " compartments.";
+    }
+};
+
+
+int main(int argc, char** argv) {
+    try {
+        bool root = true;
+        auto params = read_options(argc, argv);
+
```
[the remark this answers is not in the document]
Comment (noraabiakar) [reply example/dryrun/dryrun.cpp]: Thanks for reviewing. I implemented all requested changes.

**Comment by akuesters** [comment]
In `arbor/execution_context.cpp` line 52, could you please change 

```
execution_context::execution_context<dry_run_info>(...)
```

to 

```
execution_context::execution_context(...)
```

since it does not compile with clang MPI: 

```
/Users/kuesters/SynologyDrive/Work/97_FZJ/98_JSC/Coding/Arbor/GitHub/akuesters_arbor/arbor/execution_context.cpp:52:20: error: qualified reference to 'execution_context' is a
      constructor name rather than a type in this context
execution_context::execution_context<dry_run_info>(
                   ^
/Users/kuesters/SynologyDrive/Work/97_FZJ/98_JSC/Coding/Arbor/GitHub/akuesters_arbor/arbor/execution_context.cpp:52:37: error: expected unqualified-id
execution_context::execution_context<dry_run_info>(
                                    ^
2 errors generated.
```

**Comment by bcumming** [comment]
Hi Anne, it has already been merged, so could you please make a new PR with the fix? Nora is on vacation at the moment, so she won't be able to do it for a while.

**Comment by akuesters** [comment]
Ok, sure

**Pull request merged**